flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/ExecutorImpl.java

```diff
@@ -136,12 +147,14 @@ public ExecutorImpl(
                     "Open session to {} with connection version: {}.",
                     gatewayAddress,
                     connectionVersion);
+            Map<String, String> sessionConfig =
+                    new HashMap<>(defaultContext.getFlinkConfig().toMap());
+            sessionConfig.putAll(dynamicConfig);
```

Comment:
I think we can merge the session config into the DefaultContext. Because the session config has the same priority as DefaultContext. `DefaultContext` already exposes the API `DefaultContext#load ` to create a context with session config.